sdk/core/azure-core-perf/src/main/java/com/azure/core/perf/GetBlobProperties.java

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.perf;
+
+import com.azure.core.http.HttpClient;
+import com.azure.core.http.netty.NettyAsyncHttpClientProvider;
+import com.azure.core.perf.core.ServiceTest;
+import com.azure.core.util.HttpClientOptions;
+import com.azure.perf.test.core.PerfStressOptions;
+import com.azure.storage.blob.BlobContainerAsyncClient;
+import com.azure.storage.blob.BlobContainerClient;
+import com.azure.storage.blob.BlobServiceAsyncClient;
+import com.azure.storage.blob.BlobServiceClient;
+import reactor.core.publisher.Mono;
+
+public class GetBlobProperties extends ServiceTest<PerfStressOptions> {
+    public GetBlobProperties(PerfStressOptions options) {
+        super(options);
+    }
+
+    @Override
+    public Mono<Void> globalSetupAsync() {
+        return super.globalSetupAsync();
+    }
+
+    @Override
+    public Mono<Void> globalCleanupAsync() {
+        return super.globalCleanupAsync();
+    }
+
+    @Override
+    public void run() {
+
+        if (useSharedClient) {
+            BlobServiceClient blobServiceClient = blobServiceClientBuilder.buildClient();
+            performGetProperties(blobServiceClient);
+        } else {
+            // use a customized http client
+            HttpClientOptions clientOptions = new HttpClientOptions().setMaximumConnectionPoolSize(50);
+            HttpClient client1 = new NettyAsyncHttpClientProvider().createInstance(clientOptions);
+
+            BlobServiceClient blobServiceClient = blobServiceClientBuilder
+                .httpClient(client1)
+                .buildClient();
+
+            performGetProperties(blobServiceClient);
+        }
+    }
```

Comment:
So, given the way Storage works both of these will end up sharing the `HttpClient`. Storage passes the `HttpPipeline` of a parent resource (service client -> container client, container client -> blob client) to the child resource. It may be better to pass a `Supplier<BlobContainerClient>`, or `Function<String, BlobContainerClient>`, where the one for a shared client is passing the pre-built client and the one for a non-shared client goes through a new builder.